### PR TITLE
libgit2-glib: disable vapigen for ppc

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update `libgit2-glib` to disable `vapigen` on PPC, for the time-being. No other changes.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6 (10A190)
Xcode 3.2

### CODE REVIEW

**Review by mascguy — APPROVED** [review]
LGTM. Anyone else?

**Comment by mascguy** [comment]
@barracuda156 Have you checked the CI build logs, to see why this is failing?

**Comment by barracuda156** [comment]
> @barracuda156 Have you checked the CI build logs, to see why this is failing?

@mascguy It says:
```
../libgit2-glib-1.0.0.1/libgit2-glib/ggit-index.c:347:2: error: implicit declaration of function 'g_string_replace' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
[654](https://github.com/macports/macports-ports/runs/6068385314?check_suite_focus=true#step:8:654)
          g_string_replace (gs_path, "\\", "/", 0);
[655](https://github.com/macports/macports-ports/runs/6068385314?check_suite_focus=true#step:8:655)
          ^
[656](https://github.com/macports/macports-ports/runs/6068385314?check_suite_focus=true#step:8:656)
  ../libgit2-glib-1.0.0.1/libgit2-glib/ggit-index.c:468:2: error: implicit declaration of function 'g_string_replace' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
[657](https://github.com/macports/macports-ports/runs/6068385314?check_suite_focus=true#step:8:657)
          g_string_replace (gs_path, "\\", "/", 0);
[658](https://github.com/macports/macports-ports/runs/6068385314?check_suite_focus=true#step:8:658)
          ^
[659](https://github.com/macports/macports-ports/runs/6068385314?check_suite_focus=true#step:8:659)
  2 errors generated.
```
Should we patch out `Werror=implicit`? But this is `clang` that fails, I am not sure how it works. `gcc` builds it normally as-is.

**Comment by barracuda156** [comment]
@mascguy Okay, I know why it failed now. We should update `glib2`. I tried to re-activate an old `glib2` version, and the `libgit2-glib` failed to build for me either. I can make a PR for it.

**Comment by barracuda156** [comment]
@ryandesign We need a newer `glib2` now. Should we update `glib2-devel` and then require `libgit2-glib` to use that version specifically? Are there good reasons to be stuck at @2.64.x anyway? I have been using `glib2` @2.72.0 for the past month, and now just built @2.72.1. One patch had to be updated.
```
36-163% port -v installed glib2
The following ports are currently installed:
  glib2 @2.62.6_2+x11 requested_variants='' platform='darwin 10' archs='ppc' date='2021-10-12T13:24:54+0800'
  glib2 @2.64.6_0+x11 requested_variants='' platform='darwin 10' archs='ppc' date='2021-12-29T17:22:34+0800'
  glib2 @2.64.6_1+x11 requested_variants='' platform='darwin 10' archs='ppc' date='2022-02-19T12:42:24+0800'
  glib2 @2.72.0_0+x11 requested_variants='' platform='darwin 10' archs='ppc' date='2022-04-04T07:04:24+0800'
  glib2 @2.72.1_0+x11 (active) requested_variants='' platform='darwin 10' archs='ppc' date='2022-04-21T02:03:58+0800'
```

**Comment by mascguy** [comment]
> > @barracuda156 Have you checked the CI build logs, to see why this is failing?
> 
> @mascguy It says:
> 
> ```
> ../libgit2-glib-1.0.0.1/libgit2-glib/ggit-index.c:347:2: error: implicit declaration of function 'g_string_replace' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
> ```
> 
> Should we patch out `Werror=implicit`? But this is `clang` that fails, I am not sure how it works. `gcc` builds it normally as-is.

Per guidance from folks who are more seasoned in such areas, we generally don't want to disable that check: Implicit functions can cause crashes and other nasty behavior, particularly for ARM.

Instead, we'll want to patch those: Often it's a simple matter of a missing `#include` for a relevant header file, which needs to be added. However, I'll let others chime in, who have more experience fixing these issues.

**Comment by barracuda156** [comment]
> Per guidance from folks who are more seasoned in such areas, we generally don't want to disable that check: Implicit functions can cause crashes and other nasty behavior, particularly for ARM.
> 
> Instead, we'll want to patch those: Often it's a simple matter of a missing `#include` for a relevant header file, which needs to be added. However, I'll let others chime in, who have more experience fixing these issues.

We don’t need to disable that check, we just need to move to the newer `glib2`, then this error won’t occur. Can we make `libgit2-glib` depend on `glib2-devel` instead of `glib2`? (I assume @ryandesign won’t let us move the regular `glib2` to its latest version.)

**Comment by barracuda156** [comment]
> > > @barracuda156 Have you checked the CI build logs, to see why this is failing?
> 
> > 
> 
> > @mascguy It says:
> 
> > 
> 
> > ```
> 
> > ../libgit2-glib-1.0.0.1/libgit2-glib/ggit-index.c:347:2: error: implicit declaration of function 'g_string_replace' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
> 
> > ```
> 
> > 
> 
> > Should we patch out `Werror=implicit`? But this is `clang` that fails, I am not sure how it works. `gcc` builds it normally as-is.
> 
> 
> 
> Per guidance from folks who are more seasoned in such areas, we generally don't want to disable that check: Implicit functions can cause crashes and other nasty behavior, particularly for ARM.
> 
> 
> 
> Instead, we'll want to patch those: Often it's a simple matter of a missing `#include` for a relevant header file, which needs to be added. However, I'll let others chime in, who have more experience fixing these issues.

@mascguy Well, unfortunately updating `glib2-devel` doesn’t appear to be the option for foreseeable future: https://github.com/macports/macports-ports/pull/14637#issuecomment-1104678287

I will check later if the error can be fixed otherwise.

**Comment by mascguy** [comment]
> I will check later if the error can be fixed otherwise.

Sounds good: Assuming that this updated version isn't dependent on the latest-and-greatest API changes for `glib2` - and haven't checked to see whether it has evolved, or has mostly remained unchanged - the implicit declarations should be fixable.

**Comment by reneeotten** [comment]
port already updated to 1.1.0 in 8ce52803f6f90aee25f0daa110b8b912f0148acd

**Comment by barracuda156** [comment]
> port already updated to 1.1.0 in 8ce52803f6f90aee25f0daa110b8b912f0148acd

That still needs disabling Vapigen, but I will do it in a separate PR later.

**Comment by mascguy** [comment]
> > port already updated to 1.1.0 in [8ce5280](https://github.com/macports/macports-ports/commit/8ce52803f6f90aee25f0daa110b8b912f0148acd)
> 
> That still needs disabling Vapigen, but I will do it in a separate PR later.

You're welcome to continue using this PR. And indeed, that would be preferable to opening a new one.

Simply rebase and update everything.

**Comment by barracuda156** [comment]
@mascguy Got it, thank you.

**Comment by barracuda156** [comment]
@mascguy Rebased, done!

**Review by mascguy — APPROVED** [review]
Simple enough, I'll merge once CI completes.

**Pull request merged**
